### PR TITLE
Korean Translations

### PR DESCRIPTION
**Pull request by mojowen**
I've been working with the great folks at [Korean Resource Central LA](http://www.krcla.org/) to get a version of the guide working in Korean - not sure what's the best way to test and verify I put everything correctly in place.

I'm able to start the app locally and do some basics smoke-tests, but nothing serious. Would it make sense to try and create a test application that they can chat against in Korean and then test the output? Maybe you've got something like that already?

Let me know what I can do to get this merged in so Hello.vote can start supporting Korean - thanks!

- Scott

### CODE REVIEW

File: lib/l10n.js [hunk 1]
```diff
@@ -13,456 +13,569 @@ var l10n = {
...
         en: "Fight for the Future (FFTF), its Education Fund (FFTFEF) & {{partner}} will text you voting information and other action alerts.",
-        es: "Fight for the Future (FFTF), su Fondo de Educación (FFTFEF), y {{partner}} le enviará información de votante y otras alertas de acción."
+        es: "Fight for the Future (FFTF), su Fondo de Educación (FFTFEF), y {{partner}} le enviará información de votante y otras alertas de acción.",
+        ko: "이 앱의 주관 단체인 FFTF 및 FFTFEF (\“미래를 쟁취하자]\”) 에서 선거 정보를 포함한 여러가지 정보를 보내드릴 것입니다. "
     },
     msg_sms_fftf_stop: {
         en: "(To stop messages from FFTF & FFTFEF, text STOP.)",
-        es: "(Para detener mensajes de FFTF & FFTFEF, envíe STOP.)"
+        es: "(Para detener mensajes de FFTF & FFTFEF, envíe STOP.)",
+        ko: " 이 앱의 주관 단체인 FFTF 및 FFTFEF (\“미래를 쟁취하자]\”) 에서 선거 정보를 포함한 여러가지 정보를 보내드릴 것입니다. "
```
Comment (jlev) [inline lib/l10n.js]: The opt-out keyword needs to be STOP, not the translation into Korean. This is a legal requirement.

File: lib/l10n.js [hunk 2]
```diff
@@ -13,456 +13,569 @@ var l10n = {
...
     prompt_military_or_overseas: {
         en: "Are you a military or overseas voter? (military/overseas/no)",
-        es: "“¿Eres un votante militar o del el extranjero? (militar/extranjero/no)"
+        es: "“¿Eres un votante militar o del el extranjero? (militar/extranjero/no)",
+        ko: "군인 신분으로서 투표, 또는 미국 외 해외에서 투표를 하고자 하십니까? 군인이면 military, 해외 투표는 overseas, 둘 다 아니면 no 를 입력 해 주세요."
     },
     error_military_or_overseas: {
         en: "Sorry, I didn't get that. Please answer (military/overseas/no)",
-        es: "Lo siento, no entendí eso. Por favor contesta (militar/extranjero/no)"
+        es: "Lo siento, no entendí eso. Por favor contesta (militar/extranjero/no)"ES: "Lo siento, no entendí eso. Por favor conteste (militar/extranjero/no)"
```
Comment (jlev) [inline lib/l10n.js]: this looks malformed, please ensure there is only one key for each language

File: lib/l10n.js [hunk 3]
```diff
@@ -13,456 +13,569 @@ var l10n = {
...
         en: 'Sorry, the online voter registration deadline for {{settings.state}} has passed.',
-        es: "Lo siento, la fecha límite de inscripción para votar en {{settings.state}} ha pasado."
+        es: "Lo siento, la fecha límite de inscripción para votar en {{settings.state}} ha pasado.",
+        ko: "{{settings.state}} 주의 유권자 등록 마감일이 지났습니다."
     },
     error_validate_state_abbreviation: {
         en: 'Sorry, that\'s not a valid US state abbreviation.',
-        es: "Lo siento, eso no es un abreviatura de estado válido."
+        es: "Lo siento, eso no es un abreviatura de estado válido.",
+        ko: "그"
```
Comment (jlev) [inline lib/l10n.js]: is this the full translation of this string?

File: lib/l10n.js [hunk 4]
```diff
@@ -13,456 +13,569 @@ var l10n = {
...
         en: 'Sorry, that\'s not a valid US state name.',
-        es: "Lo siento, eso no es un nombre de estado válido."
+        es: "Lo siento, eso no es un nombre de estado válido.",
+        ko: "유효한 주 이름이 아닙니다"
     },
     msg_happy_birthday: {
         en: 'Happy birthday! \u{1F382}',
-        es: "¡Feliz cumpleaños! \u{1F382}"
+        es: "¡Feliz cumpleaños! \u{1F382}",
+        ko: "생일 축하합니다!"
```
Comment (jlev) [inline lib/l10n.js]: for consistency, please include the unicode birthday cake emoji in this message
🎂

File: lib/l10n.js [hunk 5]
```diff
@@ -13,456 +13,569 @@ var l10n = {
...
         en: 'I seem to have had a glitch. Please send your last message again.',
-        es: "Ocurrió una falla. Por favor envíe su último mensaje otra vez."
+        es: "Ocurrió una falla. Por favor envíe su último mensaje otra vez.",
+        ko: "무언가 문제가 생겼습니다. 마지막 메세지를 다시 보내주세요."
     },
     msg_error_failed: {
         en: 'Agh, an error occurred and your voter registration could not be processed. Please let us know: team@hellovote.org - reference #',
-        es: "¡Ay! Un error occurio y su inscripción para votar no pudo ser procesada. Por favor avísenos: team@hellovote.org - # de referencia  "
+        es: "¡Ay! Un error occurio y su inscripción para votar no pudo ser procesada. Por favor avísenos: team@hellovote.org - # de referencia  ",
+        ko: "유권자 등록 처리에 문제가 생겼습니다. 죄송합니다. 레퍼런스 번호 #{{reference}} 를 이메일 team@hellovote.org 으로 보내서 알려주세요"
```
Comment (jlev) [inline lib/l10n.js]: We append the reference to this message, it is not templated in.

File: lib/l10n.js [hunk 6]
```diff
@@ -474,140 +587,198 @@ var l10n = {
...
     msg_i_voted_selfie: {
-        en: "Awesome! Get your 'I VOTED' selfie badge to show your friends & get them to the polls! Visit https://i-voted.hello.vote to get yours."
+        en: "Awesome! Get your 'I VOTED' selfie badge to show your friends & get them to the polls! Visit https://i-voted.hello.vote to get yours.",
+        es: 'Estupendo! Reciba su “I VOTED” para enseñarle a sus amigos. Visita la pagina web https://i-voted.hello.vote para obtener el suyo ',
+        ko: "사합니다! 웹사이트 https://i-voted.hello.vote 에서 I Voted 셀피 뱃지를 받으시고 친구들에게도 투표를 권유하세요! "
     },
     prompt_commit_to_vote_from_notification: {
-        en: "Hey {{first_name}}, it's HelloVote. Election Day is coming up! Would you like to receive a calendar invite with your polling location?"
+        en: "Hey {{first_name}}, it's HelloVote. Election Day is coming up! Would you like to receive a calendar invite with your polling location?",
+        es: 'Hola, {{primer_nombre}} soy “HelloVote” Día de Elección se acerca! Te gustaria recibir una invitación de calendario con la ubicación del lugar a donde votar?',
```
Comment (jlev) [inline lib/l10n.js]: The templated variables need to be match the English strings. This should be `{{first_name}}`, not `{{primer_nombre}}`

File: lib/l10n.js [hunk 7]
```diff
@@ -474,140 +587,198 @@ var l10n = {
...
+        ko: "{first_name}}님, HelloVote입니다. 선거날이 다가오고 있습니다! 투표소 위치가 표시된 온라인 달력 리마인더를 보내드릴까요?"
     },
     prompt_commit_to_vote: {
-        en: "{{first_name}}, lets get you ready to vote on Tuesday, election day. Would you like to receive a calendar invite with your polling location?"
+        en: "{{first_name}}, lets get you ready to vote on Tuesday, election day. Would you like to receive a calendar invite with your polling location?",
+        es: '{{primer_nombre}], vamos a prepararte para votar el dia martes, dia de votar. Te gustaría recibir una invitación de calendario con la ubicación del lugar a donde votar?',
```
Comment (jlev) [inline lib/l10n.js]: change templated variable to `{{first_name}}`

File: lib/l10n.js [hunk 8]
```diff
@@ -617,18 +788,28 @@ var l10n = {
...
+        ko: "내일이 선거 날입니다!"
     },
     msg_election_day_today: {
         en: 'Today is Election Day!',
+        es: 'Hoy es dia de votar!',
+        ko: "오늘이 선거 날입니다!"
     },
     msg_polling_place: {
         en: 'Your polling place is at {{location}} in {{location_city}}. {{results.polling_place.link}}',
+        es: 'Su lugar de votar esta {{ubicación}} {{ubicación_cuidad}} {{results.polling_place.link}}',
```
Comment (jlev) [inline lib/l10n.js]: change to `{{location}} {{location_city}}`

File: lib/l10n.js [hunk 9]
```diff
@@ -637,10 +818,14 @@ var l10n = {
         en: 'Please enter a time, like "5pm"',
     },
     msg_gotv_1_reintro: {
-        en: "Hey {{first_name}}, it's HelloVote."
+        en: "Hey {{first_name}}, it's HelloVote.",
+        es: 'Hola {{primer_nombre}}, soy“HelloVote”',
```
Comment (jlev) [inline lib/l10n.js]: change to `{{first_name}}`

File: lib/l10n.js [hunk 10]
```diff
@@ -637,10 +818,14 @@ var l10n = {
...
+        ko: "{{first_name}}님, 안녕하세요? HelloVote 입니다."
     },
     msg_gotv_1_reintro_from_notification: {
-        en: "Hey {{first_name}}, it\'s HelloVote. Tomorrow is election day! Are you ready to vote?"
+        en: "Hey {{first_name}}, it\'s HelloVote. Tomorrow is election day! Are you ready to vote?",
+        es: 'Hola {{primer_nombre}} soy“HelloVote. Manana es dia de eleccion! Esta usted listo para votar ',
```
Comment (jlev) [inline lib/l10n.js]: change to `{{first_name}}`

File: lib/l10n.js [hunk 11]
```diff
@@ -678,15 +868,19 @@ var l10n = {
...
+        ko: "디에서 투표하셨나요? 주소, 도시, 주를 포함 해 주세요."
     },
     msg_reporting_followup: {
-        en: 'A journalist may follow up with you for more information on your story.'
+        en: 'A journalist may follow up with you for more information on your story.',
+        ko: "러분의 이야기에 대해 기자가 취재를 요청 할 수도 있습니다"
     },
     msg_thanks_for_using: {
         en: 'Thanks for using HelloVote!',
+        ko: "elloVote를 사용 해 주셔서 감사합니다"
```
Comment (jlev) [inline lib/l10n.js]: Missing an H for HelloVote?

File: lib/l10n.js [hunk 12]
```diff
@@ -474,140 +587,198 @@ var l10n = {
...
         en: 'And finally, what county was your previous address in?',
-        es: "Y por último, ¿en qué país tuvo su dirección anterior?"
+        es: "Y por último, ¿en qué país tuvo su dirección anterior?",
+        ko: "이전 주소의 도시는 어떻게 되나요?"
     },
     prompt_has_previous_name: {
         en: 'Alright then, have you been registered to vote in {{settings.state}} under a different name?',
-        es: "Entonces bueno, ha sido registrado para votar en {{settings.state}} bajo un nombre diferente?"
+        es: "Entonces bueno, ha sido registrado para votar en {{settings.state}} bajo un nombre diferente?",
+        ko: "예전에 {{setting.state}} 주 에서 다른 이름으로 유권자 등록을 한 적이 있나요?"
```
Comment (jlev) [inline lib/l10n.js]: needs to be `{{settings.state}}`

File: lib/l10n.js [hunk 13]
```diff
@@ -678,15 +868,19 @@ var l10n = {
...
+        ko: "러분의 이야기에 대해 기자가 취재를 요청 할 수도 있습니다"
     },
     msg_thanks_for_using: {
         en: 'Thanks for using HelloVote!',
+        ko: "elloVote를 사용 해 주셔서 감사합니다"
     },
     msg_disabled: {
-        en: 'Thanks for using HelloVote! This bot is now disabled!'
+        en: 'Thanks for using HelloVote! This bot is now disabled!',
+        ko: "elloVote를 사용 해 주셔서 감사합니다. 서비스가 종료되었습니다."
```
Comment (jlev) [inline lib/l10n.js]: Missing an H for HelloVote?

File: lib/l10n.js (hunk 3 above)
Comment (mojowen) [reply lib/l10n.js]: let me check with the translator

File: lib/l10n.js (hunk 2 above)
Comment (mojowen) [reply lib/l10n.js]: will reformat

File: lib/l10n.js (hunk 1 above)
Comment (mojowen) [reply lib/l10n.js]: asking for a rewrite that uses STOP in English

File: lib/l10n.js (hunk 4 above)
Comment (mojowen) [reply lib/l10n.js]: Adding

File: lib/l10n.js (hunk 5 above)
Comment (mojowen) [reply lib/l10n.js]: will ask for a retranslation with the reference number at the end

File: lib/l10n.js (hunk 12 above)
Comment (mojowen) [reply lib/l10n.js]: good catch - thank you

File: lib/l10n.js (hunk 6 above)
Comment (mojowen) [reply lib/l10n.js]: I have fixed all of these

File: lib/l10n.js (hunk 8 above)
Comment (mojowen) [reply lib/l10n.js]: updating

File: lib/l10n.js (hunk 11 above)
Comment (mojowen) [reply lib/l10n.js]: whoops - thought that was intentional but turns out I just copypaste failed. Thanks

**Comment by mojowen** [comment]
@jlev thank you so much for the review - I've sent the remaining three questions off to Yongho at KRCLA. Hope to have those cleared up soon.

**Comment by mojowen** [comment]
I've updated the remaining question with feedback from Yongho - let me know if there's anything I can address

**Pull request merged**